sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/_artifacts/_package/model_package.py

```diff
@@ -0,0 +1,292 @@
+    :type model_configuration: azure.ai.ml.entities.ModelConfiguration
+    :param tags: The tags of the model package.
+    :type tags: dict
+    """
+
+    def __init__(
+        self,
+        *,
+        name: Optional[str] = None,
+        version: Optional[str] = None,
+        inferencing_server: Optional[InferencingServer] = None,
+        base_environment_source: Optional[BaseEnvironmentSource] = None,
+        environment_variables: Optional[Dict[str, str]] = None,
+        inputs: Optional[List[ModelPackageInput]] = None,
+        model_configuration: Optional[ModelConfiguration] = None,
+        tags: Optional[Dict[str, str]] = None,
+        **kwargs,
+    ):
+        name = kwargs.pop("target_environment_name", None)
+
+        super().__init__(
+            name=name,
+            target_environment_name=name,
+            target_environment_version=version,
+            base_environment_source=base_environment_source,
+            inferencing_server=inferencing_server,
+            model_configuration=model_configuration,
+            inputs=inputs,
+            tags=tags,
+            environment_variables=environment_variables,
+            **kwargs,
+        )
+
+    @classmethod
+    def _load(
+        cls,
+        data: Optional[Dict] = None,
+        yaml_path: Optional[Union[PathLike, str]] = None,
+        params_override: Optional[list] = None,
+        **kwargs,
+    ) -> "ModelPackage":
+        params_override = params_override or []
+        data = data or {}
+        context = {
+            BASE_PATH_CONTEXT_KEY: Path(yaml_path).parent if yaml_path else Path("./"),
+            PARAMS_OVERRIDE_KEY: params_override,
+        }
+        return load_from_dict(ModelPackageSchema, data, context, **kwargs)
+
+    def dump(
+        self,
+        dest: Union[str, PathLike, IO[AnyStr]],
+        **kwargs,  # pylint: disable=unused-argument
+    ) -> None:
+        """Dump the model package spec into a file in yaml format.
+
+        :param path: Path to a local file as the target, new file will be created, raises exception if the file exists.
+        :type path: str
+        """
+        yaml_serialized = self._to_dict()
+        dump_yaml_to_file(dest, yaml_serialized, default_flow_style=False)
+
+    def _to_dict(self) -> Dict:
+        return ModelPackageSchema(context={BASE_PATH_CONTEXT_KEY: "./"}).dump(self)  # pylint: disable=no-member
+
+    @classmethod
+    def _from_rest_object(cls, model_package_rest_object: PackageResponse) -> "ModelPackageResponse":
+        target_environment_id = model_package_rest_object.target_environment_id
+        return target_environment_id
+
+    def _to_rest_object(self) -> PackageRequest:
+
+        code = None
+
+        if (
+            self.inferencing_server
+            and hasattr(self.inferencing_server, "code_configuration")
+            and self.inferencing_server.code_configuration
+        ):
+            self.inferencing_server.code_configuration._validate()
```

Comment:
What happens if validation fails? It raises an exception?

Reply:
Yes it raises an exception